micromanager_gui/main_window.py

```diff
@@ -81,6 +84,21 @@ def __init__(self, viewer: napari.viewer.Viewer, remote=False):
         self._mmc.mda.events.sequenceStarted.connect(self._on_mda_started)
         self._mmc.mda.events.sequenceFinished.connect(self._on_mda_finished)
         self._mmc.events.mdaEngineRegistered.connect(self._update_mda_engine)
+        self._mda_temp_arrays = {}
+        self._mda_temp_files: Dict[str, tempfile.TemporaryDirectory] = {}
```

Comment:
Have to keep a reference to the tempdir otherwise it gets removed and our zarr/memmap will return only zeros.

Reply:
carries the real downside of now being in charge of cleaning up files though :(